docs/architecture.md

```
@@ -84,327 +116,346 @@ graph TB
-    CRDDataProvider --> DefaultKubernetesResourceFetcher
-
-    KubernetesEntityProvider ..> BackstageLink
-    XRDTemplateEntityProvider ..> BackstageLink
-```
+### 1. Backstage Plugin Layer
 
-## Data Flow
+#### Entity Providers
 
-### 1. Initialization Flow
+**XRDTemplateEntityProvider**
```

markdownlint-cli2 (0.18.1)

Emphasis used instead of a heading

(MD036, no-emphasis-as-heading)